src/main/java/org/opensearch/knn/index/codec/nativeindex/IndexBuildHelper.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.nativeindex;
+
+import lombok.experimental.UtilityClass;
+import org.opensearch.knn.index.codec.nativeindex.model.BuildIndexParams;
+import org.opensearch.knn.index.quantizationService.QuantizationService;
+import org.opensearch.knn.index.vectorvalues.KNNVectorValues;
+import org.opensearch.knn.quantization.models.quantizationOutput.QuantizationOutput;
+import org.opensearch.knn.quantization.models.quantizationState.QuantizationState;
+
+import java.io.IOException;
+
+@UtilityClass
+class IndexBuildHelper {
```

Comment:
- This is more of a quantization related utility. Also please rethink the name, A `helper` suffix opens the possibility of misuse
- Also I did not see a unit test around this, seems like we are relying on indexbuild strategy test. Might be a good idea to have an independent unit test considering there is branching in the code

Reply:
Added Unit tests.